### PR TITLE
feat: form fields block page

### PR DESCRIPTION
**Pull request by justyna-13**
# Related issue

Closes # https://vsf.atlassian.net/jira/software/c/projects/SFUI2/boards/69?modal=detail&selectedIssue=SFUI2-641&assignee=5fcf86fca1d3f100684db8a5

# Scope of work

Form fields block page
Added email icon
Changed div wrappers to span in SfSelect

# Screenshots of visual changes
Screenshot 2023-06-06 at 13 37 25

# Checklist

- [x] Self code-reviewed
- [x] Changes documented
- [x] Semantic HTML
- [x] SSR-friendly
- [x] Caching friendly
- [x] a11y for WCAG 2.0 AA
- [x] examples created
- [x] blocks created
- [x] cypress tests created

### CODE REVIEW

**Review by AdamPawlinski — COMMENTED** [review]
Just two questions:
- `clear all` button is not working, this function will be added in a separate task, is that right? 
- shouldn't be there changeset note added if there are some, even small changes, in packages?

File: apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue [hunk 1]
```diff
@@ -0,0 +1,634 @@
+<template>
+  <div class="px-4">
+    <h1 class="mb-4 typography-headline-4 font-bold">Personal information</h1>
+    <form @submit.prevent="sendForm">
+      <label class="flex justify-between typography-text-base cursor-pointer mb-4 gap-2">
```
Comment (FRSgit) [inline apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue:5]: Is the `typography-text-base` needed here? It's the default text configuration, isn't it?

File: apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue [hunk 2]
```diff
@@ -0,0 +1,634 @@
...
+        Don’t display my personal information on a public profile
+        <SfSwitch v-model="personalInformationModelValue" />
+      </label>
+      <label>
+        <span class="typography-label-sm font-medium">Username *</span>
+        <SfInput
+          v-model="usernameModelValue"
+          :required="true"
```
Comment (FRSgit) [inline apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue:13]:
```suggestion
          required
```

File: apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue [hunk 3]
```diff
@@ -0,0 +1,634 @@
...
+          :invalid="usernameIsInvalid"
+          @input="usernameModelValue ? (usernameIsInvalid = false) : (usernameIsInvalid = true)"
```
Comment (FRSgit) [inline apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue:15]:
```suggestion
          @input="usernameIsInvalid = !usernameModelValue"
```

File: apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue [hunk 4]
```diff
@@ -0,0 +1,634 @@
...
+          @blur="usernameModelValue ? (usernameIsInvalid = false) : (usernameIsInvalid = true)"
```
Comment (FRSgit) [inline apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue:16]: Suggestion: isn't that enough?
```suggestion
          @update:model-value="usernameIsInvalid = !usernameModelValue"
```

Comment (FRSgit) [reply apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue:16]: Same for any other component

File: apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx [hunk 5]
```diff
@@ -0,0 +1,697 @@
...
+          </div>
+        </div>
+        <label
+          className={classNames('font-medium typography-label-sm block mt-4', { 'text-disabled-900': isDisabled })}
+          htmlFor={id}
+        >
+          Your Interests
+        </label>
+        <div ref={refs.setReference} className="relative">
+          <div
```
[the remark this answers is not in the document]
Comment (justyna-13) [reply apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx:569]: why input in select dropdown?

**Comment by justyna-13** [comment]
> Just two questions:
> 
> * `clear all` button is not working, this function will be added in a separate task, is that right? - Right
> * shouldn't be there changeset note added if there are some, even small changes, in packages? - I forgot about it :)

File: apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue (hunk 1 above)
Comment (justyna-13) [reply apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue]: https://github.com/vuestorefront/storefront-ui/pull/2787/commits/361d45dcea22beee7c7f07252163b9882dd2946c

File: apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue (hunk 2 above)
Comment (justyna-13) [reply apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue]: https://github.com/vuestorefront/storefront-ui/pull/2787/commits/361d45dcea22beee7c7f07252163b9882dd2946c

File: apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue (hunk 4 above)
Comment (justyna-13) [reply apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue]: https://github.com/vuestorefront/storefront-ui/pull/2787/commits/361d45dcea22beee7c7f07252163b9882dd2946c

File: apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx [hunk 6]
```diff
@@ -0,0 +1,697 @@
...
+  const [email, setEmail] = useState('');
+  const [website, setWebsite] = useState('');
+  const [areaCode, setAreaCode] = useState('');
+  const [phoneNumber, setPhoneNumber] = useState('');
+  const [description, setDescription] = useState('');
+  const [emailNotifications, setEmailNotifications] = useState(['safety-alerts']);
+  const [pushNotifications, setPushNotifications] = useState('');
+  const [searchValue, setSearchValue] = useState<string>('');
+  const [selectedValueCombobox, setSelectedValueCombobox] = useState<string>('');
+  const [isValid, setIsValid] = useState<boolean | undefined>(undefined);
```
Comment (Szymon-dziewonski) [inline apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx:207]: question: isn't default value be `undefined` so no need to define it ?
```suggestion
  const [isValid, setIsValid] = useState<boolean | undefined>();
```

File: .changeset/funny-seahorses-work.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+---
+'@storefront-ui/react': patch
+'@storefront-ui/vue': patch
+---
+
+SfSelect wrapper as span tag
```
[the remark this answers is not in the document]
Comment (justyna-13) [reply .changeset/funny-seahorses-work.md:6]: I didn't mean block :) these changes are in SfSelect https://github.com/vuestorefront/storefront-ui/pull/2787/files#diff-48fb77cf389e0906a221ab636509f56a3d9f9eb8a52245404a0d72d1f1328cfc so I think both are needed but also I may not understand something if it's wrong :)

File: apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx [hunk 8]
```diff
@@ -0,0 +1,697 @@
...
+  useEffect(() => {
+    if (searchValue && !selectedValueCombobox) {
+      const getSnippets = async () => {
+        openCombobox();
+        try {
+          const data = await mockAutocompleteRequest(searchValue);
+          setSnippets(data);
+        } catch (error) {
+          closeCombobox();
+          // eslint-disable-next-line no-console
```
Comment (Szymon-dziewonski) [inline apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx:312]: suggestion: IMHO we should place disabling comments on top of file, this piece of information is irrelevant for end user to see in source code

File: apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx [hunk 9]
```diff
@@ -0,0 +1,697 @@
...
+          console.error(error);
+        }
+      };
+      getSnippets();
+    }
+  }, [searchValue]);
+
+  const { close, toggle, isOpen } = useDisclosure({ initialValue: false });
```
Comment (Szymon-dziewonski) [inline apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx:320]:
```suggestion
  const { close, toggle, isOpen } = useDisclosure();
```
`initialValue: false` is default

File: apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx (hunk 6 above)
Comment (justyna-13) [reply apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx]: https://github.com/vuestorefront/storefront-ui/pull/2787/commits/4340777142524b2132252c34501b858df32e129f

File: apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx (hunk 8 above)
Comment (justyna-13) [reply apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx]: https://github.com/vuestorefront/storefront-ui/pull/2787/commits/4340777142524b2132252c34501b858df32e129f

**Review by Szymon-dziewonski — COMMENTED** [review]
Nice work, huge component !. I wrote only one time comments to some issues/suggestions, however those apply to react/vue

File: apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx (hunk 9 above)
Comment (justyna-13) [reply apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx]: https://github.com/vuestorefront/storefront-ui/pull/2787/commits/4340777142524b2132252c34501b858df32e129f

File: .changeset/funny-seahorses-work.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (Szymon-dziewonski) [reply .changeset/funny-seahorses-work.md:6]: you are right, sorry :)

File: apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue [hunk 10]
```diff
@@ -0,0 +1,630 @@
...
+};
+
+const countryModelValue = ref('');
+const inputRef = ref();
+const dropdownRef = ref();
+const id = useId();
+const listId = useId();
+const isDisabled = ref(false);
+const snippets = ref<{ label: string; value: string }[]>([]);
+const isValid = ref<boolean | undefined>(undefined);
```
Comment (Szymon-dziewonski) [inline apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue:522]:
```suggestion
const isValid = ref<boolean | undefined>();
```

File: apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx (hunk 8 above)
Comment (Szymon-dziewonski) [reply apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx]: could you please do this for all places, in vue as well?

File: apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue [hunk 11]
```diff
@@ -0,0 +1,630 @@
...
+      code: areaCode.value,
+      number: phoneNumberModelValue.value,
+    },
+    country: countryModelValue.value,
+    interests: selectedOption.value?.label,
+    description: descriptionValue.value,
+    emailNotifications: emailNotificationsModelValue.value,
+    pushNotifications: pushNotificationsModelValue.value,
+  };
+  // TODO: Form control
```
[the remark this answers is not in the document]
Comment (justyna-13) [reply apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue]: https://github.com/vuestorefront/storefront-ui/pull/2787/commits/fea973dd61242c6aed444743fc124bb42d16fdad

File: apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx (hunk 8 above)
Comment (justyna-13) [reply apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx]: https://github.com/vuestorefront/storefront-ui/pull/2787/commits/4340777142524b2132252c34501b858df32e129f - it was only in .tsx file

File: apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue (hunk 11 above)
[the remark this answers is not in the document]
Comment (justyna-13) [reply apps/preview/nuxt/pages/showcases/FormFields/FormFieldsRequired.vue]: added related task for textarea an dform control task is already there

File: apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx (hunk 8 above)
Comment (Szymon-dziewonski) [reply apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx]: its also in `sendForm` method https://github.com/vuestorefront/storefront-ui/pull/2787/files#diff-04928c1cb219adcd304a1db55b598f03a7052b732e3917fbe41d8b45debe4c43R371

Comment (justyna-13) [reply apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx]: right :) https://github.com/vuestorefront/storefront-ui/pull/2787/commits/64f5cd0ac9fe5d2c33c5b245badcf7367f4379db

**Review by Szymon-dziewonski — APPROVED** [review]
Looking good from my side :) gj

File: apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (AdamPawlinski) [reply apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx:568]: My bad, I thought it is Combobox block :) and that is why the arrow key navigation doesn't work. So there is probably need to add separate `useTrapFocus`  for this `selectDropdown`  block to make it work, wdyt?

[the remark this answers is not in the document]
Comment (AdamPawlinski) [reply apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx:568]: formfields

[the remark this answers is not in the document]
Comment (justyna-13) [reply apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx:568]: what do you think about creating a new bug? This task is quite big and to be honest it should be fixed in SelectDropdown itself :)

[the remark this answers is not in the document]
Comment (justyna-13) [reply apps/preview/next/pages/showcases/FormFields/FormFieldsRequired.tsx:569]: https://github.com/vuestorefront/storefront-ui/pull/2787/commits/f15677c23ee028d56bea4749727f74bd9e2579f7

**Comment by AdamPawlinski** [comment]
Looks great, excellent work :+1:

**Review by AdamPawlinski — APPROVED** [review]

**Review by FRSgit — APPROVED** [review]

**Pull request merged**